sdk/core/System.ClientModel/samples/ModelReaderWriter.md

```diff
@@ -53,4 +53,35 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same `IPersistableModel<T>` or `IJsonModel<T>` and register it with the `ModelReaderWriterOptions`.
```

Comment:
It would be helpful to give a "for example" here  that captured the following information:

For example, if a user wanted to how `InputModel` was read and written, they would create a "proxy type" that implements the `IPersistableModel<InputModel>` interface.  The implementation of `IPersistableModel<InputModel>.Create` will override how ModelReaderWriter.Read deserializes the model data into a runtime instance of `InputModel` -- i.e. it will use this method instead of `InputModel`'s own `Create` implementation.  Similarly, the implementation of `IPersistableModel<InputModel>.Write` in the proxy type will override how ModelReaderWrite.Write serializes the model data.